### ISSUE TITLE
[Bug]: Address CVEs affecting server.jar and node pkg

### ISSUE THREAD

**Issue by FraPazGal** [appsmithorg/appsmith#23365]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

**What did you see?**

When running a vulnerabilities scan using Trivy in latest version of appsmith, `v1.9.19`, it flags several CVEs as follows:

```
Java (jar)

Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 1, CRITICAL: 1)

┌─────────────────────────────────┬────────────────┬──────────┬───────────────────┬───────────────┬───────────────────────────────────────────────────────────┐
│             Library             │ Vulnerability  │ Severity │ Installed Version │ Fixed Version │                           Title                           │
├─────────────────────────────────┼────────────────┼──────────┼───────────────────┼───────────────┼───────────────────────────────────────────────────────────┤
│ com.h2database:h2 (server.jar)  │ CVE-2022-45868 │ HIGH     │ 2.1.210           │               │ The web-based admin console in H2 Database Engine through │
│                                 │                │          │                   │               │ 2.1.214 can ...                                           │
│                                 │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-45868                │
├─────────────────────────────────┼────────────────┼──────────┼───────────────────┼───────────────┼───────────────────────────────────────────────────────────┤
│ org.yaml:snakeyaml (server.jar) │ CVE-2022-1471  │ CRITICAL │ 1.33              │ 2.0           │ Constructor Deserialization Remote Code Execution         │
│                                 │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-1471                 │
└─────────────────────────────────┴────────────────┴──────────┴───────────────────┴───────────────┴───────────────────────────────────────────────────────────┘

Node.js (node-pkg)

Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 0)

┌──────────────────────────┬────────────────┬──────────┬───────────────────┬───────────────┬────────────────────────────────────────────┐
│         Library          │ Vulnerability  │ Severity │ Installed Version │ Fixed Version │                   Title                    │
├──────────────────────────┼────────────────┼──────────┼───────────────────┼───────────────┼────────────────────────────────────────────┤
│ engine.io (package.json) │ CVE-2023-31125 │ MEDIUM   │ 6.2.1             │ 6.4.2         │ engine.io Uncaught Exception vulnerability │
│                          │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-31125 │
└──────────────────────────┴────────────────┴──────────┴───────────────────┴───────────────┴────────────────────────────────────────────┘
```

**What did you expect to see?**

No CVEs.

**Additional information**

Per [H2Database developers](https://github.com/h2database/h2database/issues/3686), the vulnerability is related to a corner-case passing the password through CLI, which is not recommended. Seeing they do not consider this a CVEs and thus won't be fixing it, the question here is whether Appsmith is affected by this vulnerability.

There is already a [dependabot PR](https://github.com/appsmithorg/appsmith/pull/23231) to update `engine.io`, so the fix on this should be trivial.

### Steps To Reproduce

```
$ trivy --version
Version: 0.41.0
$ trivy image --vuln-type library index.docker.io/appsmith/appsmith-ce:v1.9.19
```

### Environment

Release

### Version

Appsmith CE - 1.9.19

**Comment by nidhi-nair** [comment]
This has been fixed in #23572, the latest version should not be seeing the SnakeYAML CVE anymore.

**Comment by LucasDesenv** [comment]
Hi @nidhi-nair , thanks for the fix. In which Spring Boot version we'll see this change?
